question22/main.js

```diff
@@ -0,0 +1,188 @@
+'use strict';
+const wrapper = document.getElementById('js-wrapper');
+
+const createElementWithClassName = (element, name) => {
+  const createdElement = document.createElement(element);
+  createdElement.classList.add(name);
+  return createdElement;
+};
+
+const showLoadImg = () => {
+  const loadImg = createElementWithClassName("img", "loading");
+  loadImg.src = "./img/loading-circle.gif";
+  loadImg.id = 'js-loading';
+  wrapper.appendChild(loadImg);
+} 
+
+const removeLoadImg = () => document.getElementById('js-loading').remove();
+
+const getData = new Promise((resolve) => {
+  showLoadImg();
+  setTimeout(() => {
+    resolve(fetch('https://mocki.io/v1/5381547b-04ff-4a89-b169-dd84425f5196'));
+  }, 3000);
+})
+
+//Get json data
+const callApi = async() => {
+  try {
+    const response = await getData;
+    if (!response.ok) {
+      console.error(`${response.status}:${response.statusText}`);
+    }
+    const json = await response.json();
+    return json.data;
+  }
+  catch(error) {
+    console.error(error);
+  }
+  finally {
+    removeLoadImg();
+  }
+}
+
+const init = async() => {
+  const userData = await callApi();
+  if (userData){
+    renderTableElement(userData);
+    } else {
+    wrapper.textContent = "Failed to display data.";
+    console.error('Failed to display data.'); 
+  }
+}
+init();
+
+const userDataColumn = {
+  userId: {
+    value : "ID",
+    sort : true
+  },
+  name: {
+    value : "名前",
+    sort : false
+  },
+  gender: {
+    value : "性別",
+    sort : false
+  },
+  age: {
+    value : "年齢",
+    sort : true
+  }
+}
+
+const renderTableElement = (userData)  => {
+  const table = createElementWithClassName("table", "table");
+  const tbody = createElementWithClassName("tbody", "tbody");
+  table.id = "js-table";
+  tbody.id = "js-tbody";
+  wrapper.appendChild(table).appendChild(getTableHeadElement());
+  wrapper.appendChild(table).appendChild(tbody).appendChild(getTableRowFragment(userData));
+  sortButtonClickEvent(userData);
+}
+
+const getTableHeadElement = () => {
+  const thead = createElementWithClassName("thead", "thead");
+  const tr = createElementWithClassName("tr", "tr");
+  const fragmentThElement = document.createDocumentFragment();
+  const userDataColumnKeys = Object.keys(userDataColumn);
+
+  userDataColumnKeys.forEach((key) => {
+    const th = createElementWithClassName("th", "th");
+    const value = userDataColumn[key].value;
+    th.textContent = value; 
+    userDataColumn[key].sort && th.appendChild(createSortButton());
```

Comment:
The addition of the sort button has been changed to be specified in the userDataColumn object.